server/src/main/java/org/opensearch/identity/PrincipalIdentifierToken.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.opensearch.common.io.stream.NamedWriteable;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Requester Token for requests to/from an extension
+ * Each user will have different token for different extension
+ */
+public class PrincipalIdentifierToken implements Serializable, NamedWriteable {
```

Comment:
I don't think there's any value in implementing Serializable, and I've read multiple places it should not be used in new code, and only used when required to interact with legacy code.

Its removal from Java is a [long term goal](https://adtmag.com/articles/2018/05/30/java-serialization.aspx).

Reply:
was originally on the fence of keeping, but will remove it!